ipa-core/src/protocol/context/dzkp_validator.rs

```diff
@@ -843,261 +910,150 @@ mod tests {
-            let mut vec_y_right = Vec::<u8>::new();
-            let mut vec_prss_left = Vec::<u8>::new();
-            let mut vec_prss_right = Vec::<u8>::new();
-            let mut vec_z_right = Vec::<u8>::new();
-
-            // gen 1024 random values
-            for _i in 0..1024 {
-                let x_left: u8 = rng.gen();
-                let x_right: u8 = rng.gen();
-                let y_left: u8 = rng.gen();
-                let y_right: u8 = rng.gen();
-                let prss_left: u8 = rng.gen();
-                let prss_right: u8 = rng.gen();
-                let z_right: u8 = rng.gen();
-
-                // fill expected
-                expected_x_left.push(Fp31::truncate_from(x_left));
-                expected_x_right.push(Fp31::truncate_from(x_right));
-                expected_y_left.push(Fp31::truncate_from(y_left));
-                expected_y_right.push(Fp31::truncate_from(y_right));
-                expected_prss_left.push(Fp31::truncate_from(prss_left));
-                expected_prss_right.push(Fp31::truncate_from(prss_right));
-                expected_z_right.push(Fp31::truncate_from(z_right));
-
-                // fill segment vec
-                vec_x_left.push(x_left);
-                vec_x_right.push(x_right);
-                vec_y_left.push(y_left);
-                vec_y_right.push(y_right);
-                vec_prss_left.push(prss_left);
-                vec_prss_right.push(prss_right);
-                vec_z_right.push(z_right);
-            }
+            let vec_x_left = (0..1024).map(|_| rng.gen::<u8>()).collect::<BitVec<u8>>();
+            let vec_x_right = (0..1024).map(|_| rng.gen::<u8>()).collect::<BitVec<u8>>();
+            let vec_y_left = (0..1024).map(|_| rng.gen::<u8>()).collect::<BitVec<u8>>();
+            let vec_y_right = (0..1024).map(|_| rng.gen::<u8>()).collect::<BitVec<u8>>();
+            let vec_prss_left = (0..1024).map(|_| rng.gen::<u8>()).collect::<BitVec<u8>>();
+            let vec_prss_right = (0..1024).map(|_| rng.gen::<u8>()).collect::<BitVec<u8>>();
+
+            // compute z
+            let vec_z_left = vec_x_left.clone() & vec_y_left.clone()
+                ^ (vec_x_left.clone() & vec_y_right.clone())
+                ^ (vec_x_right.clone() & vec_y_left.clone())
+                ^ vec_prss_left.clone()
+                ^ vec_prss_right.clone();
+
+            // vector for unchecked elements
+            // i.e. these are used to fill the segments of the verifier and prover that are not part
+            // of this povers proof
+            let vec_z_right = (0..1024).map(|_| rng.gen::<u8>()).collect::<BitVec<u8>>();
+            let vec_x = (0..1024).map(|_| rng.gen::<u8>()).collect::<BitVec<u8>>();
+            let vec_y = (0..1024).map(|_| rng.gen::<u8>()).collect::<BitVec<u8>>();
+            let vec_prss = (0..1024).map(|_| rng.gen::<u8>()).collect::<BitVec<u8>>();
+            let vec_z = (0..1024).map(|_| rng.gen::<u8>()).collect::<BitVec<u8>>();
```

Comment:
These aren't the best names. These are really confusingly close to the other names. 

Suggestion: either name this something like `x_unused_in_proof` or `x_3rd_share`.